starter/app/src/main/java/com/udacity/project4/locationreminders/savereminder/selectreminderlocation/SelectLocationFragment.kt

```diff
@@ -544,12 +555,15 @@ class SelectLocationFragment : BaseFragment(), OnMapReadyCallback {
         )
     }
 
+    /**
+     * When the user confirms on the selected location, send back the selected location details
+     * to the view model and navigate back to the previous fragment to save the reminder and add
+     * the geofence.
+     */
```

Comment:
Hello @islamalsawy .

Let me explain to you what happened here.

This PR was created 3 weeks ago with the intention to fix the starter code of project [Build a Location Reminder App](https://learn.udacity.com/nanodegrees/nd940/parts/cd0638/lessons/53b98c03-1fe9-4535-95f1-104ac0ccdc36/concepts/f094aa61-69ee-4288-ae63-3b71ef5c219e).

3 weeks ago the only commit to be merged with [udacity:master](https://github.com/udacity/nd940-android-kotlin-c4-starter) was [this commit](https://github.com/udacity/nd940-android-kotlin-c4-starter/pull/12/commits/99b1b59a6e82d5e71cffd740b96ac5cdec17d793), that is the starter commit for students to work on:

**Code updated to run on SDK 33 and Gradle 7.5**

Since then I keep implementating this project because I needed to check how deprecated Udacity's suggestions were on [Advanced Android with Kotlin](https://www.udacity.com/course/advanced-android-with-kotlin--ud940) course. There are several deprecations here. All the following concepts are deprecated, and Udacity haven't updated this in the course:

- **Notifications**, that now need Runtime Permissions and need newer Flags to be able to run.
- **Geofences**, that were incorrectly using **JobIntentService** (that doesn't work anymore since 2021), and now need to be implemented using **WorkManager**.
- **onRequestPermission** method, that was deprecated by Google, and now need to be replaced by **registerForActivityResult**.
- **onCreateOptionsMenu** and **onOptionsItemSelected** methods, that are now deprecated by Google, and need to be replaced by **MenuHost.addMenuProvider**.
- **onActivityResult** method, that is now deprecated by Google, and need to be replaced by **registerForActivityResult**.
- Also, the **Location Tracking** feature using GPS in Android that was never taught in this Udacity course. This feature is requested on project [Build a Location Reminder App](https://learn.udacity.com/nanodegrees/nd940/parts/cd0638/lessons/53b98c03-1fe9-4535-95f1-104ac0ccdc36/concepts/c95c9510-e130-4055-8bbf-acfeaa1c63ca), and in the last years the implementation of this feature in Android changed dramatically.

Any student enrolled on this course, to have this final project finished, will need to fix all those deprecations! I think this is a very heavy load for them to carry, right?

We have a lot to fix on this course!!

Anyway, to begin with, what I suggest here is that I close this PR and create a newer one with **only** the starter commit (as explained earlier). This way future students will be able, at least, to start this project with an updated starter code, containing all the correct Gradle dependencies and SDK versions up to date.

If you need further explanations, please contact me in private on Udacity Circle website.

Reply:
New PR created:
https://github.com/udacity/nd940-android-kotlin-c4-starter/pull/14

Reply:
@graffiti75 Thank you for putting things in context. Let's start with approving this new PR. I am going to share more details with you Circle.